src/main/java/slimeknights/tconstruct/library/recipe/casting/AbstractCastingRecipe.java

```diff
@@ -79,4 +96,19 @@ public boolean matches(ICastingInventory inv, World worldIn) {
   public NonNullList<Ingredient> getIngredients() {
     return NonNullList.from(Ingredient.EMPTY, this.cast);
   }
+
+  /**
+   * Cooling time for the recipe.
+   * @return Cooling time specified in the json, else a dynamically calculated one.
+   */
+  public int getCoolingTime() {
```

Comment:
Instead of calculating this on the fly always, have a recipe `getCoolingTime(ICastingInventory)`